src/ZiraLink.Api.Application/Tools/HttpTools.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.NetworkInformation;
+using System.Text;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Xml.Linq;
+
+namespace ZiraLink.Api.Application.Tools
+{
+    public class HttpTools : IHttpTools
+    {
+        private readonly IHttpClientFactory _httpClientFactory;
+        public HttpTools(IHttpClientFactory httpClientFactory)
+        {
+            _httpClientFactory = httpClientFactory; 
+        }
+
+        public async Task<bool> CheckDomainExists(string domainUrl)
+        {
+            var httpClient = _httpClientFactory.CreateClient();
+            try
+            {
+                using (var response = await httpClient.GetAsync(domainUrl, HttpCompletionOption.ResponseHeadersRead))
+                {
+                    response.EnsureSuccessStatusCode();
+                    var stream = await response.Content.ReadAsStreamAsync();
+                } 
+            }
+            catch (Exception)
+            { 
+            }
+
+            return await PingDomain(domainUrl);
+             
+        }
+        
+        public async Task<bool> PingDomain(string domainUrl)
+        {
+            domainUrl = domainUrl.Replace("https://", "").Replace("http://", "");
+            try
+            {
+                Ping ping = new Ping();
+                for (int i = 0; i < 2; i++)
+                {
+                    var response = await Task.Run(() => ping.Send(domainUrl, 1200));
```

Comment:
Replace ping.Send with ping.SendAsync

Reply:
I've used SendPingAsync. But it didn't work. The response is only "TimeOut".
![image](https://github.com/saeedmaghdam/ZiraLink.Api/assets/49972383/b252eb58-8a19-47a7-ad4f-e45215b2c3ae)

Could you give me a suggestion?

Reply:
It worked 👍